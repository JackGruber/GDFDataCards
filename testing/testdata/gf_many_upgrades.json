{
    "id": "PBJIbTjbQb9m",
    "gameSystem": "gf",
    "armyId": "oqnnu0gk8q6hyyny",
    "armyIds": [
        "oqnnu0gk8q6hyyny"
    ],
    "armyFaction": "Prime Brothers",
    "armyName": "Prime Brothers",
    "modified": "2024-04-01T18:03:10.657Z",
    "saveVersion": 3,
    "favourite": false,
    "armyVersions": [
        {
            "version": "3.2.0",
            "armyId": "oqnnu0gk8q6hyyny"
        }
    ],
    "listPoints": 285,
    "list": {
        "id": "PBJIbTjbQb9m",
        "key": "BqSHHgU838apps0WiBUkL",
        "name": "Prime Brothers",
        "description": "",
        "pointsLimit": 2000,
        "campaignMode": false,
        "narrativeMode": false,
        "forceOrg": true,
        "gameSystem": "gf",
        "isCloud": false,
        "modified": "2024-04-01T18:03:10.658Z",
        "cloudModified": null,
        "activationCount": 1,
        "modelCount": 5,
        "units": [
            {
                "id": "81WTJ34",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "O8JFx",
                "selectedUpgrades": [
                    {
                        "instanceId": "tcq4TA2Ax",
                        "upgradeId": "ar5sR",
                        "optionId": "wyKaPTZ"
                    },
                    {
                        "instanceId": "_Fwk3C6YA",
                        "upgradeId": "tbNwc",
                        "optionId": "uNapO"
                    },
                    {
                        "instanceId": "-oGFLnBhT",
                        "upgradeId": "tbNwc",
                        "optionId": "uNapO"
                    },
                    {
                        "instanceId": "sOYy-x6yu",
                        "upgradeId": "Fyjqd",
                        "optionId": "dOV9k"
                    },
                    {
                        "instanceId": "NPJ-j3DoH",
                        "upgradeId": "vIcBJVF",
                        "optionId": "lCeyzzU"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            }
        ]
    }
}