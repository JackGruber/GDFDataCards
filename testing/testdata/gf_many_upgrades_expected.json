{
    "armyId": "oqnnu0gk8q6hyyny",
    "gameSystem": "gf",
    "gameSystemId": 2,
    "armyName": "Prime Brothers",
    "listPoints": 285,
    "listName": "Prime Brothers",
    "units": [
        {
            "type": "Desolator Squad",
            "name": "Desolator Squad",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "81WTJ34",
            "cost": 210,
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "D1",
                "ftzK4"
            ],
            "size": 5,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 5,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 2,
                    "count": 2,
                    "name": "HE-Launcher",
                    "range": 30,
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Vengeance HE-Launcher",
                    "range": 30,
                    "specialRules": [
                        {
                            "key": "blast",
                            "name": "Blast",
                            "type": "ArmyBookRule",
                            "label": "Blast(3)",
                            "modify": false,
                            "rating": "3"
                        },
                        {
                            "key": "indirect",
                            "name": "Indirect",
                            "type": "ArmyBookRule",
                            "label": "Indirect",
                            "modify": false,
                            "rating": ""
                        }
                    ]
                },
                {
                    "attacks": 1,
                    "count": 2,
                    "name": "AT-Launcher",
                    "range": 30,
                    "specialRules": [
                        {
                            "key": "lock-on",
                            "name": "Lock-On",
                            "type": "ArmyBookRule",
                            "label": "Lock-On",
                            "modify": false,
                            "rating": ""
                        }
                    ],
                    "ap": "2"
                },
                {
                    "attacks": 1,
                    "count": 5,
                    "name": "Mini-GL",
                    "range": 18,
                    "specialRules": [
                        {
                            "key": "indirect",
                            "name": "Indirect",
                            "type": "ArmyBookRule",
                            "label": "Indirect",
                            "modify": false,
                            "rating": ""
                        },
                        {
                            "name": "Limited",
                            "type": "ArmyBookRule",
                            "label": "Limited",
                            "modify": false
                        }
                    ]
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "medical-training",
                    "name": "Medical Training",
                    "type": "ArmyBookRule",
                    "label": "Medical Training",
                    "rating": "",
                    "count": 1
                }
            ],
            "upgradeCost": [
                20,
                5,
                5,
                15,
                30
            ]
        }
    ]
}